operator/builtin/output/otlp/otlp_test.go

```diff
@@ -0,0 +1,161 @@
+package otlp
+
+import (
+	"context"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/observiq/stanza/entry"
+	"github.com/observiq/stanza/operator/helper"
+	"github.com/observiq/stanza/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOTLPConfigBuild(t *testing.T) {
+	t.Run("OutputConfigError", func(t *testing.T) {
+		cfg := NewOTLPOutputConfig("test")
+		cfg.OperatorType = ""
+		_, err := cfg.Build(testutil.NewBuildContext(t))
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "missing required `type` field")
+	})
+
+	t.Run("MissingEndpoint", func(t *testing.T) {
+		cfg := NewOTLPOutputConfig("test")
+		cfg.Endpoint = ""
+		_, err := cfg.Build(testutil.NewBuildContext(t))
+		require.Error(t, err)
+		require.Equal(t, err.Error(), "'endpoint' is required")
+	})
+
+	t.Run("InvalidEndpoint", func(t *testing.T) {
+		cfg := NewOTLPOutputConfig("test")
+		cfg.Endpoint = `%^&*($@)`
+		_, err := cfg.Build(testutil.NewBuildContext(t))
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "is not a valid URL")
+	})
+}
+
+func TestOTLPOutput(t *testing.T) {
+	cases := []struct {
+		name   string
+		cfgMod func(*OTLPOutputConfig)
+		input  []*entry.Entry
+	}{
+		{
+			"Simple",
+			nil,
+			[]*entry.Entry{{
+				Timestamp: time.Date(2016, 10, 10, 8, 58, 52, 0, time.UTC),
+				Record:    "test",
+			}},
+		},
+		{
+			"Multi",
+			nil,
+			[]*entry.Entry{{
+				Timestamp: time.Date(2016, 10, 10, 8, 58, 52, 0, time.UTC),
+				Record:    "test1",
+			}, {
+				Timestamp: time.Date(2016, 10, 10, 8, 58, 52, 0, time.UTC),
+				Record:    "test2",
+			}},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ln := newListener()
+			addr, err := ln.start()
+			require.NoError(t, err)
+			defer ln.stop()
+
+			cfg := NewOTLPOutputConfig("test")
+			cfg.Endpoint = addr
+			cfg.TLSSetting.Insecure = true
+			cfg.FlusherConfig.MaxWait = helper.NewDuration(time.Millisecond)
+			if tc.cfgMod != nil {
+				tc.cfgMod(cfg)
+			}
+
+			ops, err := cfg.Build(testutil.NewBuildContext(t))
+			require.NoError(t, err)
+			op := ops[0]
+			require.NoError(t, op.Start())
+			for _, entry := range tc.input {
+				require.NoError(t, op.Process(context.Background(), entry))
+			}
+			defer op.Stop()
+
+			// TODO test connection?
+			// expectTestConnection(t, ln)
+			expected, err := Convert(tc.input).ToOtlpProtoBytes()
+			require.NoError(t, err)
+			expectRequestBody(t, ln, expected)
+		})
+	}
+}
+
+func expectRequestBody(t *testing.T, ln *listener, expected []byte) {
+	select {
+	case body := <-ln.requestBodies:
+		require.Equal(t, expected, body)
+	case <-time.After(time.Second):
+		require.FailNow(t, "Timed out waiting for test connection")
+	}
+}
+
+type listener struct {
+	server        *http.Server
+	requestBodies chan []byte
+}
+
+func newListener() *listener {
+	requests := make(chan []byte, 100)
+
+	mux := http.NewServeMux()
```

Comment:
No need for a mux if you just have a single endpoint

Reply:
Per our discussion, we'll just leave this.